lib/core/util.js

```diff
@@ -26,6 +26,57 @@ function isBlobLike (object) {
   )
 }
 
+function encode (val) {
+  return encodeURIComponent(val).replace(/%3A/gi, ':').replace(/%24/g, '$').replace(/%2C/gi, ',').replace(/%20/g, '+').replace(/%5B/gi, '[').replace(/%5D/gi, ']')
+}
```

Comment:
What does this do? Can we add a comment explaining?

Reply:
Whole buildURL block is simplified adaptation of how Axios is doing it. I can add tests for when this is actually used as well as a comment, including the reference to source.